File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> res;
        unordered_map<string,vector<string>> m;
        for(string& s:strs){
            string t = s;
            sort(t.begin(),t.end());
            m[t].push_back(s);
        }
        for(auto& [k,vs]:m){
            res.push_back(vs);
        }
        return res;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        if(strs==null || strs.length==0)return new ArrayList<>();
        HashMap<String,List<String>> m =  new HashMap<>();
        for(String s:strs){
            char[] ca = new char[26];
            for(char c:s.toCharArray())ca[c-'a']++;
            String key = String.valueOf(ca);
            if(!m.containsKey(key))m.put(key,new ArrayList<>());
            m.get(key).add(s);
        }
        return new ArrayList<>(m.values());
    }
}